internal/driver/webhtml.go

```diff
@@ -0,0 +1,555 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package driver
+
+import "html/template"
+
+var graphTemplate = template.Must(template.New("graph").Parse(
+	`<!DOCTYPE html>
+<html>
+<head>
+<meta charset="utf-8"/>
+<title>{{.Title}}</title>
+<style type="text/css">
+html, body {
+  height: 100%;
+  min-height: 100%;
+  margin: 0px;
+}
+body {
+  width: 100%;
+  overflow: hidden;
+}
+h1 {
+  font-weight: normal;
+  font-size: 24px;
+  padding: 0em;
+  margin-top: 5px;
+  margin-bottom: 5px;
+}
+#page {
+  display: flex;
+  flex-direction: column;
+  height: 100%;
+  min-height: 100%;
+  width: 100%;
+  min-width: 100%;
+  margin: 0px;
+}
+#header {
+  flex: 0 0 auto;
+  width: 100%;
+}
+#leftbuttons {
+  float: left;
+}
+#rightbuttons {
+  float: right;
+  display: table-cell;
+  vertical-align: middle;
+}
+#rightbuttons label {
+  vertical-align: middle;
+}
+#scale {
+  vertical-align: middle;
+}
+#graph {
+  flex: 1 1 auto;
+  overflow: hidden;
+}
+svg {
+  width: 100%;
+  height: auto;
+  border: 1px solid black;
+}
+button {
+  margin-top: 5px;
+  margin-bottom: 5px;
+}
+#reset, #scale {
+  margin-left: 10px;
+}
+#detailtext {
+  display: none;
+  position: absolute;
+  background-color: #ffffff;
+  min-width: 160px;
+  border-top: 1px solid black;
+  box-shadow: 2px 2px 2px 0px #aaa;
+  z-index: 1;
+}
+</style>
+</head>
+<body>
+<h1>{{.Title}}</h1>
+<div id="page">
+
+<div id="errors">{{range .Errors}}<div>{{.}}</div>{{end}}</div>
+
+<div id="header">
+<div id="leftbuttons">
+<button id="details">&#x25b7; Details</button>
+<div id="detailtext">
+{{range .Legend}}<div>{{.}}</div>{{end}}
+</div>
+<button id="list">List</button>
+<button id="disasm">Disasm</button>
+<input id="searchbox" type="text" placeholder="Search regexp" autocomplete="off" autocapitalize="none" size=40/>
+<button id="focus">Focus</button>
+<button id="ignore">Ignore</button>
+<button id="hide">Hide</button>
+<button id="show">Show</button>
+<button id="reset">Reset</button>
+</div>
+<div id="rightbuttons">
+</div>
+</div>
+
+<div id="graph">
+{{.Svg}}
+</div>
+
+</div>
+</body>
+</html>
+<script>
+
+// Make svg pannable and zoomable.
+// Call clickHandler(t) if a click event is caught by the pan event handlers.
+function initPanAndZoom(svg, clickHandler) {
```

Comment:
function statements are used here for global functions, but assignment of function expressions are used for the inner functions, is there any reason?
Unless reassigning a variable with another function, the statements have the advantage of being order independent and can be called before they are defined in the source, so the code can be top-down. The assignments have to be in a correct order and defined before used (unless used in another function and called later), which enforced bottom-up order).